### DISCUSSION TITLE
Share store across different SPA microfrontends

### DISCUSSION THREAD

**Discussion by francisco-alonso** [vuejs/pinia#2451] [Help and Questions]
Hello,

Im trying to serve a Pinia store from an external module so I can reuse it in each micrfrontend served by Single SPA. I have two questions:

- Is there any lib mode to export the store ?
- How we can prevent state unstabilities when multiple components (sspa frontends) are mutating the state?

Thanks

**Comment by posva** [comment]
- Make sure you mark pinia and vue as external, they cannot be bundled with your lib
- You don't need to

**Reply by francisco-alonso** [reply]
Gracias @posva !

So, to make sure I follow your answer:

- In my root config of single SPA, I have to configure as peer deps both vue and pinia. So, when I import my lib I will be able to use my store in all my mfes.
- I put you an example: let's say I have a modalStore to manage the modal state and I share it across all microfrontends, you tell me there's no additional config as one modal will be displayed and not all mfes will try to open the modal (overlapping modals). They dont share the same ref, is that correct?
